### ISSUE TITLE
Add 'oc-env' command which display information to eval the oc binary to PATH

### ISSUE THREAD

**Issue by hferentschik** [minishift/minishift#142]
See issue #141 

> We can also offer something similar to docker-env, say oc-env which will output something which is evaluable in the shell and updates the PATH in order to execute oc

**Comment by LalatenduMohanty** [comment]
+1, this will be super useful for OC binary users.

**Comment by budhrg** [comment]
Also, along with adding cached oc binary into PATH, display other information can be useful
```
# You can access the OpenShift console on: https://10.1.2.2:8443/console
# To use OpenShift CLI, run: oc login https://10.1.2.2:8443
export OPENSHIFT_URL=https://10.1.2.2:8443
export OPENSHIFT_WEB_CONSOLE=https://10.1.2.2:8443/console
export DOCKER_REGISTRY=hub.openshift.centos7-adb.10.1.2.2.xip.io
``` 

Taken from [ADB doc](https://github.com/projectatomic/adb-atomic-developer-bundle/blob/master/docs/using.adoc#configure-your-environment)

**Comment by hferentschik** [comment]
Actually we might want to create a _oc_ subcommand instead, allowing for potentially other _oc_ related commands. See also issue #349

**Comment by budhrg** [comment]
Related to #465

**Comment by budhrg** [comment]
@hferentschik @praveenkumar @LalatenduMohanty 

So here we need to display OS specific instructions to be evaluated in shell.

I am bit concern how we handle Windows specific instructions. Eval is not available as per this https://github.com/minishift/minishift/issues/782 so should we just display the instructions and tell user to run it ?

**Comment by hferentschik** [comment]
I don't understand why the problem is. I might be missing something. But 'docker-env' is already doing all the Os and shell specific stuff - https://github.com/minishift/minishift/blob/7974f5edc9697c6b878794aa9f98d9d0fe668588/cmd/minishift/cmd/env.go#L66-L66. So 'oc-env' would do the equivalent, and yes, the command to change the PATH needs to be different depending on OS and shell. 

From a ux point of view, we do the same as with 'docker-env'. The user runs 'minishift oc-env'. This will determine OS and shell and print the right command for the user to configure his shell with the modified PATH. The user then cut&paste this command into the shell.

Where exactly is the problem?

**Comment by hferentschik** [comment]
#782 is a documentation issue. The user is not supposed to directly run 'eval ...' directly. He runs 'minishift docker-env' which will print the right command to configure the variables. Of course, once the user knows what he is doing, he might for example type directly _Invoke-Expression mininishift docker-env_ or _Invoke-Expression mininishift oc_env_.

**Comment by hferentschik** [comment]
I always envisioned that we extract the code for determining OS and shell type into some utility class and then reuse this bit for docker-env and oc-env. There is plenty of code which can be shared there. I guess one needs to refactor a fair bit here to make the code sharable. ATM everything is quite baked in into _env.go_. I think that's the biggest challenge here. 

Also, we probably want to have the _forceShell_ option as well.

**Comment by hferentschik** [comment]
> Also, along with adding cached oc binary into PATH, display other information can be useful

-1, this is just about adding _oc_ to the PATH.

**Comment by hferentschik** [comment]
Also, we need to make sure that we add the _oc_ path, to  the front of the current PATH.

**Comment by budhrg** [comment]
> I always envisioned that we extract the code for determining OS and shell type into some utility class and then reuse this bit for docker-env and oc-env. There is plenty of code which can be shared there. I guess one needs to refactor a fair bit here to make the code sharable

Make sense in extracting the common code here 👍

**Comment by budhrg** [comment]
> Also, we probably want to have the forceShell option as well.

@hferentschik , what is the intention here?

**Comment by hferentschik** [comment]
The shell detection algorithm is fragile and in some cases can be wrong. So even if you shell supports a specific syntax you cannot get the right output, since the detection fails. The force flags allows you to enforce a certain syntax if you know it works for your shell.

**Comment by hferentschik** [comment]
Merged via pull request #853
